### ISSUE TITLE
Strength measure for any opponent (suggestion)

### ISSUE THREAD

**Issue by petgo3** [gcp/leela-zero#2115]
In combination with #2057 the difference to optimal move value could be collected (only sum it).
Perhaps the sum of differences (to the optimal move) / movecount is related to the game quality.
With #2057 this might be easy to implement ...

**Comment by Splee99** [comment]
Maybe this is a bit off topic, but I'm also following the strength of Levels zero, same as you. What I observed from the recent weights, is that it sometimes plays just yose moves in the middle of the game. I guess it is because of a heavy value head. Currently the cfg_puct has a fixed value of 0.8. I'm doing some experiment to see if 0.9 is better.

**Comment by Friday9i** [comment]
@Ishinoshita Humm, either I misunderstood something, or it is easy to solve that issue of search policy vs winrate ; -)!
To evaluate the quality of the opponent's move n, we should not use the view of LZ at move n-1 (as move n may have received little visits, or even no visit at all) but the winrate of LZ when playing move n+1, ie after it was properly studied by LZ.
Hence, after LZ plays move n-1, if the opponent plays an unexpected move n (for which LZ allocated no visits when playing move n-1), LZ will of course study the possible answers to this move n (eg 1600 visits for selfplay games) and hence, it will get a reasonable and deep opinion about it: the winrate will be quite relevant (unless the opponent is much stronger and winrate after 1600 visits is still not good).
Did I overlooked something?

**Comment by Ishinoshita** [comment]
@Friday9i Yes, you're fully right, for an indicator based on win rate delta, no need to do anything else, data already there. My remark would apply only to an indicator based on visits (could be more a 'matter of style' indicator, as more focused on disagreement with policy / bot own preference).
